src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -165,6 +170,40 @@ def enable_autocomplete(self):
         argcomplete.autocomplete(self, validator=lambda c, p: c.lower().startswith(p.lower()),
                                  default_completer=lambda _: ())
 
+    @classmethod
+    def _make_failure_recovery_recommendations(cls, *args, **kwargs):
+        if telemetry.is_telemetry_enabled():
+            cls.recommendation_provider(telemetry._get_core_version(), *args, **kwargs)  # pylint: disable=protected-access
```

Comment:
To get core version, you may simply use https://github.com/Azure/azure-cli/blob/34f9033b96e6ea31f82626a9f2b152c1a061300e/src/azure-cli-core/azure/cli/core/util.py#L804

instead of relying on telemetry.

Reply:
Switched to suggested method of getting core version.